### DISCUSSION TITLE
rAudio 20240719: Fixed and non-fixed bugs

### DISCUSSION THREAD

**Discussion by klassikmann** [rern/rAudio#1714] [Fixed Issues]
The main issue described in https://github.com/rern/rAudio/discussions/1708 (Spurious library view as filesystem path) has been **fixed**.

The issue mentioned in the same thread:
> Additional issue with UPDATE on desktop browsers without swiping support: Help guide bitmaps doesn't scale anymore.

is fixed also.

Now to the **non-fixed** (or new) issue:
> Using desktop browsers (Windows, Linux) without swiping support there is no way to switch to the library or playback view by clicking.

But there is a (awkward) workaround in clicking on thy symbol in the upper left corner:
![pic1](https://github.com/user-attachments/assets/7e4e1915-861b-4696-8d6f-10637169ee2b)

**EDIT**:
If you apply the workaround twice the former behaviour is reinstated.

**Comment by ceejayemm** [comment]
@klassikmann 

I have found that access to the Library and Playlist screens (on non-swipeable web browsers etc) can be achieved by clicking the top left icon (square within a square) which makes the bottom bar available even when the playback screen has no bar visible.

Chris

**Reply by klassikmann** [reply]
@ceejayemm 
Your tip is - I assume - the same as the workaround I outlined above. Applying the workaround twice the bottom bar vanished again without losing the preferred former behaviour 
- click at the botton in the left side of the browser window = switch to library view
- click at the botton in the right side of the browser window = switch to playlist view
- click at the botton in the middle of the browser window = switch to playback view

**Reply by ceejayemm** [reply]
I guess so.  I wasn't originally sure about 'without losing the preferred former behaviour' and 'If you apply the workaround twice the former behaviour is reinstated.'. I am not sure I understand what the 'former behaviour' is that you refer to.

Chris

**Reply by klassikmann** [reply]
@ceejayemm 

> I am not sure I understand what the 'former behaviour' is that you refer to.

The 'former behaviour' up to release 20240707 is the ability just to click at the bottom of the browser window to switch between the three views (library, playlist, playback). To use the funtionality in 20240719 I have to apply the workaround twice. 
Background: I'm using rAudio in the playback view in "coverart only" mode.

**Reply by ceejayemm** [reply]
Ah, I see what you mean.  After the second click of the icon the bottom bar is visibly turned off but it still seems to be there to click on.  Its as if the page is remembering the bar activities even though its not showing the bar.

Chris

**Reply by klassikmann** [reply]
@ceejayemm 
Exactly. As a consequnce after a refresh (F5) of the page rAudio has forgotten the bar activities. Therefore I classified the new behaviour as a bug.

**Reply by Lightwel** [reply]
Good spot thanks.  However, I am finding that when I re-enable the bottom row buttons, this also enables the top row (not an issue for me) but both rows do not now auto-hide.  Also, I cannot now find the menu (gearwheel, top right corner) items to control this behaviour.  For the time being I can use the albumart-only controls, but it would be nice to have the previous behaviour fully reinstated. R

**Reply by Lightwel** [reply]
Actually partly my mistake.  The bars are controllable via the Playback menu, except that they are always on, and deselecting the ¨bars always on´ option does not cure this.
R

**Comment by tjengbudi** [comment]
i am using this version. and i cannot open sd card and usb song list. anyone that have workaround to make it works? is it a bug?

**Reply by R101G** [reply]
rAudio is behaving very strangely for me. If I play an album and then go to Library, it still shows the previous album played, rather than the one currently in the playlist. There are other odd things occurring as well, but not every time.

With so many setting options available, it seems that everyone is seeing different problems, and fixing one problem causes another.

**Reply by tjengbudi** [reply]
@R101G how you add the file? save in sdcard? or in the nas? what the expected behavior of click sd icon in library??

**Comment by klassikmann** [comment]
@R101G 
I have a similar impression. @rern changed/redesigned so many PHP- and JS-scripts in the last time that I'm not surprised to see an unusually large number of regressions. Nevertheless I'm confident that starting with the next release a pure bug hunting (= no new features) will take place.

**Comment by R101G** [comment]
I do wonder whether, if possible, a more modular approach would be better, with new features presented as add-ons, rather than integrated into the package alongside a bug fix.

**Comment by klassikmann** [comment]
> ...
> ...
> Now to the **non-fixed** (or new) issue:
> 
> > Using desktop browsers (Windows, Linux) without swiping support there is no way to switch to the library or playback view by clicking.
> 
> But there is a (awkward) workaround in clicking on thy symbol in the upper left corner: ![pic1](https://private-user-images.githubusercontent.com/25383513/350280944-7e4e1915-861b-4696-8d6f-10637169ee2b.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM5NjU0NzQsIm5iZiI6MTcyMzk2NTE3NCwicGF0aCI6Ii8yNTM4MzUxMy8zNTAyODA5NDQtN2U0ZTE5MTUtODYxYi00Njk2LThkNmYtMTA2MzcxNjllZTJiLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE4VDA3MTI1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUwZmUzYzNmMDZhNzk0ODBhZDZkZTI4OWZmOWFkMzIwNjVjZWMzY2IwYzdiZTc2MDYyZmY3YWYzZmE4OGY0MzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AbrTX_2S2kEddrs8gwcHZFqmM3lokdybpqPuOYJodUg)
> 
> **EDIT**: If you apply the workaround twice the former behaviour is reinstated.

The issue has been fixed in release 20240818.
